#207 Map Obstacles

Ben4566 opened the pull request:
## Changelog

### TileMapLayer Objects
- Changed `z_index` to `1`
- Added collisions to the tileset tree trunks

### rider_enemy
- Changed `collision_mask` to `8`

### mounted_enemy
- Changed `collision_mask` to `8`

### ranged_enemy
- Changed `collision_mask` to `8`

### default_player
- Changed `collision_mask` of **Archer**, **Assassin**, **Knight**, and **Mage** to `8`

### default_enemy
- Changed `collision_mask` to `8`

### Arrow
- Changed `collision_mask` to `8`

### Mage
- Changed `collision_mask` to `8`

### Knight
- Changed `collision_mask` to `8`

### Assassin
- Changed `collision_mask` to `8`

### Archer
- Changed `collision_mask` to `8`

### bow_arrow
- Changed `collision_mask` to `8`

### crossbow_arrow
- Changed `collision_mask` to `8`

### kunai_projectile
- Changed `collision_mask` to `8`

## Notes

- Currently, the enemies do not seem to have any visible issues navigating around obstacles. Therefore, I suggest we close **issue #211: Enemies Navigation around Obstacles** for now. However, please **do not delete the issue**, as we might implement more complex obstacles in the future where navigation becomes problematic again.

- At the moment, both players and enemies can move behind bushes. I'm not sure whether this should be changed or left as is — just putting this out there as something to think about.

manuel0804 commented:
pls do not name the branches starting with a '#', this causes problems when checking out the branch in Command line.
I would recommand to create branches in issues over the button "create branch for this issue".

manuel0804 reviewed (changes requested):
1. The wave timer and current wave counter are affected as well by the obstacles - they should be visable the whole time.
2. When the character is in front of the tree and you move towards the tree, the player is behind
![image](https://github.com/user-attachments/assets/cc636b8e-cfc3-4780-9d82-aea430f73a9c)
3. For the bushes i would say we remove the whole thing (or if not too hard: check wheather the x-axis is at a specific point of the bush (e.g. the centre point) and based on that decide if player is in front or behind bush)

manuel0804 commented:
so do you change something now or not? i didnt got it :/

Ben4566 commented:
I did fixed the points 1 and 3

manuel0804 commented:
i think you forgot to push your commits, for me there are no changes and point 1 and 3 are still not fixed. @Ben4566

manuel0804 commented:
pls merge master 
and sorry for the misunderstanding: I didn't mean "remove the bushes" but "remove the animation for the bushes" - if its not too complex you can insert them again, otherwise just let me now, then we merge without them

Ben4566 commented:
I’ve removed the bushes for now and added two new layers: one for tree trunks and one for leaves.

It would be possible to re-add the bushes, but the assets are still just placeholders. Re-adding them wouldn’t really change the process of creating a new map, since creating a new map would involve deleting all existing tile positions anyway — meaning the map creator would repaint everything from scratch.
